riff-raff/app/persistence/TargetDynamoRepository.scala

```diff
@@ -0,0 +1,38 @@
+package persistence
+
+import ci.Target
+import com.amazonaws.services.dynamodbv2.model.PutItemResult
+import com.gu.scanamo.Table
+import org.joda.time.DateTime
+
+case class TargetId(targetKey: String, projectName: String, region: String, stack: String, app: String, lastSeen: DateTime) {
+  def matches(target: Target): Boolean = region == target.region && stack == target.stack && app == target.app
+}
+object TargetId {
+  def apply(tgt: Target, projectName: String, lastSeen: DateTime): TargetId =
+    TargetId(targetKey(tgt), projectName, tgt.region, tgt.stack, tgt.app, lastSeen)
+  def targetKey(tgt: Target) = Seq(tgt.region,tgt.stack,tgt.app).mkString("|")
+}
+
+object TargetDynamoRepository extends DynamoRepository {
+  def tablePrefix = "riffraff-target-ids"
+  val table = Table[TargetId](tableName)
+
+  import com.gu.scanamo.syntax._
+
+  def set(target: Target, projectName: String, lastSeen: DateTime): PutItemResult =
+    exec(table.put(TargetId(target, projectName, lastSeen)))
+
+  def get(targetKey: String, projectName: String): Option[TargetId] = {
+    exec(table.get('targetKey -> targetKey and 'projectName -> projectName)).flatMap(_.toOption)
+  }
+
+  def get(target: Target): List[TargetId] = {
```

Comment:
(nit) I'd prefer if this had a different name, rather than being overloaded

Reply:
`get`, `Target` or `TargetId`?

Reply:
Oh - `get` is overloaded now isn't it. OK, I'll have a look.